hedera-node/docs/design/services/smart-contract-service/frictionless-airdrops-system-contracts.md

```diff
@@ -0,0 +1,151 @@
+and their respective redirect function calls (`cancelAirdropFT`, `cancleAirdropNFT`, `claimAirdropFT`, `claimAirdropNFT`, `rejectTokenFT`, `rejectTokenNFTs`, `setAutomaticAssociations`) that will extend the capabilities of the Hedera Smart Contract Service (HSCS) to support frictionless airdrops.
+
+## References
+
+[HIP-904](https://hips.hedera.com/hip/hip-904) - HIP that introduces the frictionless airdrops.
+
+## Goals
+
+- Expose `airdropToken`, `claimAirdrops`, `cancelAirdrops` and `rejectTokens` as new functions in the Hedera Token Service Smart Contract.
+- Expose `cancelAirdropFT`, `cancleAirdropNFT`, `claimAirdropFT`, `claimAirdropNFT`, `rejectTokenFt`, `rejectTokenNFTs`, `setAutomaticAssociations` as new functions in the proxy redirect token facade contract IHRC.
+- Implement the needed HTS system contract classes to support the new functions.
+
+## Non Goals
+
+- The implementation of the HAPI operation, as it is already an existing feature.
+
+## Architecture
+
+The architecture for the frictionless airdrops follows the existing framework defined for handling all calls to the HederaTokenService system contract in the modularization services and is described in more detail in the Implementation section below.
+
+## Implementation
+
+### New Solidity Structures
+
+We will introduce the following new structures to support the new functionality:
+
+`PendingAirdrop` - A struct that represents a pending airdrop request.
+
+```solidity
+struct PendingAirdrop {
+    address sender;
+    address receiver;
+
+    address token;
+    bool isNft;
+    int64 serial;
+}
+```
+
+`NftId` - A struct that represents the Nft serials to be rejected.
+
+```solidity
+struct NftId {
+    address nft;
+    int64[] serials;
+}
+```
+
+### New Solidity Functions
+
+New system contract functions must be added to the `IHederaTokenService` interface to support airdropping tokens.
+
+| Function Selector Hash |                                                 Function Signature                                                  |    Response    |                                 |
+|------------------------|---------------------------------------------------------------------------------------------------------------------|----------------|---------------------------------|
+| `0x2f348119`           | `function airdropTokens(TokenTransferList[] memory tokenTransfers) external returns (int64 responseCode)`           | `ResponseCode` | The response code from the call |
+| `0xc1acfe5d`           | `function cancelAirdrops(PendingAirdrop[] memory pendingAirdrops) external returns (int64 responseCode)`            | `ResponseCode` | The response code from the call |
+| `0x241b6af7`           | `function claimAirdrops(PendingAirdrop[] memory pendingAirdrops) external returns (int64 responseCode)`             | `ResponseCode` | The response code from the call |
```

Comment:
Here (and in next table) the HIP limits the number of pending airdrops to a maximum of 10.  That limit should be in this document somewhere - perhaps in this table, perhaps as text following the table?